src/main/java/graphql/execution/FutureValueUnboxer.java

```diff
@@ -0,0 +1,22 @@
+package graphql.execution;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+public class FutureValueUnboxer implements ValueUnboxer {
+
+    @Override
+    public Object unbox(Object object, ValueUnboxingContext context) {
+        if (object instanceof Future) {
+            try {
+                return context.unbox(((Future<?>) object).get());
```

Comment:
This is a BIG no right here - we must never called `Future.get` - this makes execution synchrnous and graphql-java is aiming to be asynch in all its aspects.